File: src/1260.py
# DFS와 BFS
from collections import defaultdict, deque
N, M, V = map(int, input().split())
graph = defaultdict(set)
for i in range(M):
    a, b = map(int, input().split())
    graph[a].add(b)
    graph[b].add(a)

dfs_visited = defaultdict(bool)
dfs_way = []
def dfs(x):
    if dfs_visited[x]: return
    dfs_visited[x] = True
    dfs_way.append(x)
    for i in graph[x]:
        dfs(i)
dfs(V)

def bfs(start):
    queue = deque()
    visited = defaultdict(bool)
    queue.append(start)
    visited[start] = True
    res = []
    while queue:
        node = queue.popleft()
        res.append(node)
        for i in graph[node]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True
    return res

print(*dfs_way)
print(*bfs(V))

File: src/1260.js
let n=require('fs').readFileSync('dev/stdin').toString().trim().split('\n');
let [N, M, V] = n.shift().split(' ').map(v=>+v);
let arr = Array.from(Array(N+1), () => Array(N+1).fill(0));
let visited = Array(N+1).fill(false);
while (n.length) {
  let [a, b] = n.shift().split(' ').map(v=>+v);
  arr[a][b] = 1;
  arr[b][a] = 1;
}
let dfs = [];
function DFS(idx) {
  dfs.push(idx)
  for (let i = 1; i <= N; i++) {
    if (arr[idx][i] && !visited[i]) {
      visited[i] = true;
      DFS(i)
    }
  }
}

visited[V] = true;
DFS(V);
console.log(dfs.join` `);

visited = visited.map(() => false);
let queue = [];
let bfs = [];
function BFS(idx) {
  visited[idx] = true;
  queue.push(idx);
  while(queue.length) {
    idx = queue.shift();
    bfs.push(idx);
    for (let i = 1; i <= N; i++) {
      if (arr[idx][i] && !visited[i]) {
        visited[i] = true;
        queue.push(i);
      }
    }
  }  
}

BFS(V);
console.log(bfs.join(` `));
